analysis/model/04_01_(b)_cox_format_survival_data.R

```diff
@@ -233,19 +233,11 @@ fit_get_data_surv <- function(event,subgroup, stratify_by_subgroup, stratify_by,
     
     # Calculate median person-time -----------------------------------------------
     
-    tmp <- data_surv[,c("patient_id","days_cat","tstart","tstop","cox_weights")]
+    tmp <- data_surv[data_surv$event ==1,c("patient_id","days_cat","tstart","tstop","cox_weights")]
     tmp$person_time <- tmp$tstop - tmp$tstart
     
-    tmp_post_expo <- tmp %>% filter(days_cat != 0) %>% 
-      group_by(patient_id) %>%
-      summarise(person_time = sum(person_time)) %>%
-      left_join(tmp %>% select(patient_id, cox_weights) %>% distinct(), by = "patient_id") %>%
-      mutate(days_cat = max(episode_info$days_cat)) %>%
-      select(- patient_id)
-    
     tmp[,c("patient_id","tstart","tstop")] <- NULL
-    tmp <- rbind(tmp,tmp_post_expo)
-    
+
     tmp <- tmp %>%
       dplyr::group_by(days_cat) %>%
       dplyr::mutate(median_follow_up = matrixStats::weightedMedian(x = person_time, w = cox_weights)) %>%
```

Comment:
I think you can use a normal median now (instead of weighted median) because everyone with an outcome should have a weight of 1 in the analysis.

Reply:
https://github.com/opensafely/post-covid-vaccinated/pull/308/commits/377fda19a809ffd5f57412159dd25808b105d1a1